### PR TITLE
Followup to remove inherited Brave Ads clang tidy options

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/28022

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/choose/eligible_ads_predictor_util.h [hunk 1]
```diff
@@ -54,6 +54,16 @@ CreativeAdPredictorMap<T> GroupCreativeAdsByCreativeInstanceId(
   return creative_ad_predictors;
 }
 
+template <typename T>
+std::vector<T> SetIntersection(std::vector<T> lhs, std::vector<T> rhs) {
```
Comment (goodov) [inline vendor/bat-native-ads/src/bat/ads/internal/ads/serving/choose/eligible_ads_predictor_util.h:58]: looks more like `VectorIntersection`.

**Review by goodov — APPROVED** [review]

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/choose/eligible_ads_predictor_util.h (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads/serving/choose/eligible_ads_predictor_util.h]: Good catch, thanks and fixed

File: browser/brave_ads/device_id/device_id_impl.cc [hunk 2]
```diff
@@ -65,7 +65,7 @@ bool IsValidMacAddressImpl(const void* bytes, size_t size) {
   constexpr size_t kOuiLength = 3;
 
   // VPN, virtualization, tethering, bluetooth, etc.
-  static MacAddressInfo invalid_mac_addresses[] = {
+  static const MacAddressInfo kInvalidMacAddresses[] = {
```
Comment (aseren) [inline browser/brave_ads/device_id/device_id_impl.cc:68]: this should be changed to static constexpr

File: components/brave_ads/content/browser/search_result_ad/search_result_ad_parsing_unittest.cc [hunk 3]
```diff
@@ -75,7 +75,7 @@ class TestWebPageConstructor final {
   }
 
   schema_org::mojom::EntityPtr CreateCreativeEntity() {
-    const char* search_result_ad_string_attributes[] = {
+    const char* kSearchResultAdStringAttributes[] = {
```
Comment (aseren) [inline components/brave_ads/content/browser/search_result_ad/search_result_ad_parsing_unittest.cc:78]: this should be: `static constexpr const char*`

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/exclusion_rules/conversion_exclusion_rule.cc [hunk 4]
```diff
@@ -20,9 +20,8 @@ constexpr int kConversionCap = 1;
...
 bool DoesRespectCap(const AdEventList& ad_events,
                     const CreativeAdInfo& creative_ad) {
-  const int count = std::count_if(
-      ad_events.cbegin(), ad_events.cend(),
-      [&creative_ad](const AdEventInfo& ad_event) {
+  const size_t count = base::ranges::count_if(
+      ad_events, [&creative_ad](const AdEventInfo& ad_event) {
         return ad_event.confirmation_type == ConfirmationType::kConversion &&
                ad_event.creative_set_id == creative_ad.creative_set_id;
       });
```
Comment (aseren) [inline vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/exclusion_rules/conversion_exclusion_rule.cc:27]: we can convert 
`constexpr int kConversionCap = 1;` to
`constexpr size_t kConversionCap = 1;`

File: vendor/bat-native-ads/src/bat/ads/internal/common/search_engine/search_engine_subdomain_constants.cc [hunk 5]
```diff
@@ -11,7 +11,7 @@ namespace ads {
 
 const std::vector<std::string>& GetWikipediaSearchEngineSubdomains() {
   // See https://www.wikipedia.org/.
-  static base::NoDestructor<std::vector<std::string>> extensions(
+  static const base::NoDestructor<std::vector<std::string>> kExtensions(
```
Comment (aseren) [inline vendor/bat-native-ads/src/bat/ads/internal/common/search_engine/search_engine_subdomain_constants.cc:14]: This should be converted to `static constexpr char* kExtensions[]`

File: vendor/bat-native-ads/src/bat/ads/internal/covariates/covariate_manager.cc [hunk 6]
```diff
@@ -29,83 +29,90 @@ using UserActivityEventToCovariateTypesMapping =
...
-       {UserActivityEventType::kTypedKeywordOtherThanDefaultSearchProvider,
-        {brave_federated::mojom::CovariateType::
-             kNumberOfTypedKeywordOtherThanDefaultSearchProviderEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastTypedKeywordOtherThanDefaultSearchProviderEvent}},
-       {UserActivityEventType::kTypedUrl,
-        {brave_federated::mojom::CovariateType::kNumberOfTypedUrlEvents,
-         brave_federated::mojom::CovariateType::kTimeSinceLastTypedUrlEvent}}});
-  return *mappings;
+  static const base::NoDestructor<UserActivityEventToCovariateTypesMapping>
```
Comment (aseren) [inline vendor/bat-native-ads/src/bat/ads/internal/covariates/covariate_manager.cc:35]: Need to consider converting to fixed_flat_map

File: vendor/bat-native-ads/src/bat/ads/internal/processors/behavioral/purchase_intent/purchase_intent_processor.cc [hunk 7]
```diff
@@ -56,17 +55,10 @@ KeywordList ToKeywords(const std::string& value) {
...
-  KeywordList sorted_keywords_lhs = keywords_lhs;
-  std::sort(sorted_keywords_lhs.begin(), sorted_keywords_lhs.end());
-
-  KeywordList sorted_keywords_rhs = keywords_rhs;
-  std::sort(sorted_keywords_rhs.begin(), sorted_keywords_rhs.end());
-
-  return std::includes(sorted_keywords_lhs.cbegin(), sorted_keywords_lhs.cend(),
-                       sorted_keywords_rhs.cbegin(),
-                       sorted_keywords_rhs.cend());
+bool IsSubset(KeywordList keywords_lhs, KeywordList keywords_rhs) {
```
Comment (aseren) [inline vendor/bat-native-ads/src/bat/ads/internal/processors/behavioral/purchase_intent/purchase_intent_processor.cc:58]: check if arguments moving is possible

File: vendor/bat-native-ads/src/bat/ads/internal/common/search_engine/search_engine_subdomain_constants.cc (hunk 5 above)
Comment (goodov) [reply vendor/bat-native-ads/src/bat/ads/internal/common/search_engine/search_engine_subdomain_constants.cc:14]: agree. no need to all these static arrays use runtime memory.

File: vendor/bat-native-ads/src/bat/ads/internal/covariates/covariate_manager.cc (hunk 6 above)
Comment (goodov) [reply vendor/bat-native-ads/src/bat/ads/internal/covariates/covariate_manager.cc:35]: please do

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/exclusion_rules/conversion_exclusion_rule.cc (hunk 4 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/exclusion_rules/conversion_exclusion_rule.cc:27]: I have another follow up PR to fix this throughout brave ads

File: vendor/bat-native-ads/src/bat/ads/internal/common/search_engine/search_engine_subdomain_constants.cc (hunk 5 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/common/search_engine/search_engine_subdomain_constants.cc:14]: @aseren will fix in his related PR

File: vendor/bat-native-ads/src/bat/ads/internal/covariates/covariate_manager.cc (hunk 6 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/covariates/covariate_manager.cc:35]: @aseren will fix in his related PR

File: vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/exclusion_rules/conversion_exclusion_rule.cc (hunk 4 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads/serving/eligible_ads/exclusion_rules/conversion_exclusion_rule.cc:27]: Changed to int as will fix types in another PR

File: vendor/bat-native-ads/src/bat/ads/internal/processors/behavioral/purchase_intent/purchase_intent_processor.cc (hunk 7 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/processors/behavioral/purchase_intent/purchase_intent_processor.cc:58]: Fixed

File: components/brave_ads/content/browser/search_result_ad/search_result_ad_parsing_unittest.cc (hunk 3 above)
Comment (tmancey) [reply components/brave_ads/content/browser/search_result_ad/search_result_ad_parsing_unittest.cc:78]: Fixed

**Pull request merged**